src/analysis/event_detector.py

```diff
@@ -0,0 +1,312 @@
+import xarray as xr
+import numpy as np
+from pathlib import Path
+from xclim.run_length import rle, longest_run  # , windowed_run_events
+from typing import Tuple, Optional, Any
+import warnings
+import pandas as pd
+from scripts.eng_utils import get_ds_mask
+
+
+class EventDetector():
+    def __init__(self,
+                 path_to_data: Path,
+                 data_folder: Path = Path('data')) -> None:
+        self.data_folder = data_folder
+        self.interim_folder = data_folder / "interim"
+        self.processed_folder = data_folder / "processed"
+
+        if not self.processed_folder.exists():
+            self.processed_folder.mkdir()
+
+        assert path_to_data.exists(), f"{path_to_data} does not point to an existing file!"
+        self.ds = self.read_data(path_to_data)
+
+    def read_data(self, path_to_data: Path) -> xr.Dataset:
+
+        try:
+            ds = xr.open_dataset(path_to_data)
+
+        except ValueError:
+            print(ValueError)
+            warnings.warn("Having to decode_times=False because unsupported calendar")
+            ds = xr.open_dataset(path_to_data, decode_times=False)
+            warnings.warn("Hardcoding MONTHLY data (CHIRPS example)")
+            time = pd.date_range(start='1900-01-01', freq='M', periods=len(ds.time.values))
```

Comment:
Why is this starting in 1900? The data doesn't start then

Reply:
Why are the time values not just pulled from the CHIRPS data itself?

Reply:
because sometimes when there is a non-standard calendar it is necessary to set the `decode_times=False` and then recode the time. Tbh it's a really hacky solution specific to ONE specific dataset. So probably best to avoid that and just tell the user to do it htemselves